lib/bank_accounts/checking_account.rb

```diff
@@ -0,0 +1,64 @@
+module BankAccounts
+  class CheckingAccount < Account
+    attr_reader :checks_used
+
+    def initialize(id, initial_balance)
+      @id = id
+      @checks_used = 0
+
+      # must have at least $0 balance
+      if initial_balance < 0
+        raise ArgumentError, "Checking account can't have a negative balance. Create a new account."
+      else
+        @balance = initial_balance
+      end
+    end
+
+
+    def withdraw(amount)
+      amount = amount.abs # real ATMs don't have a 'negative number' button
+      fee = 1
```

Comment:
I think if you did this - you could potentially generalize that to the overall account class and then update them for every subclass. Then you wouldn't have to repeat as much code (per your PR comment)